ec2/spark_ec2.py

```diff
@@ -750,7 +754,7 @@ def setup_cluster(conn, master_nodes, slave_nodes, opts, deploy_ssh_key):
                'mapreduce', 'spark-standalone', 'tachyon']
 
     if opts.hadoop_major_version == "1":
-        modules = filter(lambda x: x != "mapreduce", modules)
+        modules = list(filter(lambda x: x != "mapreduce", modules))
```

Comment:
I didn't know how all the machinery in this script worked, so I was hesitant to make anything but minimal changes.  I saw that there was a desire to move it over to argparse?

I also have a major linguistic nit to pick with the asymmetry between launch/destroy.